### PR TITLE
feat: add BigCommerce example

### PR DESCRIPTION
**Pull request by deini**
### BigCommerce + Makeswift Example

This PR includes a catalyst storefront bootstrapped with Catalyst's CLI with some modifications to run Makeswift marketing pages.

`/app` -> Catalyst
`/pages` -> Makeswift

### CODE REVIEW

**Review by chanceaclark — DISMISSED** [review]
Most of this is catalyst, and that part looks good on my end.

**Comment by deini** [comment]
@migueloller are you guys hardcoding `next build` on vercel? We have our own `build` script that runs codegen first. The codegen requires the BC env var to be set in vercel too.

Without codegen thats why we are getting all this import errors.

PS: I'll also remove the lockfile, leftover from copying the standalone project into the monorepo.

**Comment by migueloller** [comment]
> @migueloller are you guys hardcoding `next build` on vercel? We have our own `build` script that runs codegen first. The codegen requires the BC env var to be set in vercel too.
> 
> Without codegen thats why we are getting all this import errors.
> 
> PS: I'll also remove the lockfile, leftover from copying the standalone project into the monorepo.

Yeah we are. I can get the fixed for you. I'll need that env var, too.

**Review by migueloller — APPROVED** [review]
I didn't look at any of the code that's no Makeswift related. The stuff that's Makeswift specific looks good to me!

Have a couple of minor comments and some questions.

File: examples/bigcommerce/README.md
[the remark this answers is not in the document]
Comment (deini) [reply examples/bigcommerce/README.md:1]: The diff is collapsed but there is a README https://github.com/makeswift/makeswift/blob/0a0c9e861449714aa733305e8a738affa19e8769/examples/bigcommerce/README.md

Or do you want something extra added to this README?

**Comment by migueloller** [comment]
It seems that Vercel isn't installing development dependencies, which is unexpected. Also, the command it says it's running is `pnpm install`, not `pnpm install --production` 🤔 

Any ideas here @deini?

```sh
Running "pnpm run build"
19:27:36.761 |  
19:27:36.761 | > bigcommerce@0.1.0 build /vercel/path0/examples/bigcommerce
19:27:36.761 | > npm run codegen && next build
19:27:36.761 |  
19:27:37.198 |  
19:27:37.198 | > bigcommerce@0.1.0 codegen
19:27:37.198 | > dotenv -e .env.local -- gql-gen --errors-only --config codegen.ts
19:27:37.198 |  
19:27:37.206 | sh: dotenv: command not found
19:27:37.215 | ELIFECYCLE  Command failed.
19:27:37.235 | Error: Command "pnpm run build" exited with 1
19:27:37.906
```

**Comment by migueloller** [comment]
> PS: I'll also remove the lockfile, leftover from copying the standalone project into the monorepo.

I think we'll probably want to keep the lockfile, no?

**Comment by migueloller** [comment]
> It seems that Vercel isn't installing development dependencies, which is unexpected. Also, the command it says it's running is `pnpm install`, not `pnpm install --production` 🤔
> 
> Any ideas here @deini?
> 
> ```shell
> Running "pnpm run build"
> 19:27:36.761 |  
> 19:27:36.761 | > bigcommerce@0.1.0 build /vercel/path0/examples/bigcommerce
> 19:27:36.761 | > npm run codegen && next build
> 19:27:36.761 |  
> 19:27:37.198 |  
> 19:27:37.198 | > bigcommerce@0.1.0 codegen
> 19:27:37.198 | > dotenv -e .env.local -- gql-gen --errors-only --config codegen.ts
> 19:27:37.198 |  
> 19:27:37.206 | sh: dotenv: command not found
> 19:27:37.215 | ELIFECYCLE  Command failed.
> 19:27:37.235 | Error: Command "pnpm run build" exited with 1
> 19:27:37.906
> ```

I think the problem here might be that it's detecting pnpm from the root lockfile and then deciding to use is instead of npm.

File: examples/bigcommerce/README.md
[the remark this answers is not in the document]
Comment (migueloller) [reply examples/bigcommerce/README.md:1]: Sorry, the diff looked like the entire file was deleted. That README is great!

CleanShot 2024-02-15 at 19 39 07

File: examples/bigcommerce/.env.local.example
[the remark this answers is not in the document]
Comment (migueloller) [reply examples/bigcommerce/.env.local.example:1]: I cleaned these up. If there's any that are missing we can add them back.

**Comment by joshuawootonn** [comment]
> > PS: I'll also remove the lockfile, leftover from copying the standalone project into the monorepo.
> 
> I think we'll probably want to keep the lockfile, no?

@migueloller is this supposed to be used in our CLI? I am curious why you would want to keep the lock file.

**Comment by joshuawootonn** [comment]
RIP to the `next-pwa` stuff that was an absolute mud wrestle to add to this example. I was not impressed with `next-pwa` though so it's for the better!

File: examples/bigcommerce/tailwind.config.js [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
+      animation: {
+        revealVertical: 'revealVertical 400ms forwards cubic-bezier(0, 1, 0.25, 1)',
       },
     },
   },
-  plugins: [],
-}
+  // @ts-ignore
+  // eslint-disable-next-line global-require
+  plugins: [require('tailwindcss-radix')(), require('tailwindcss-animate')],
```
Comment (joshuawootonn) [inline examples/bigcommerce/tailwind.config.js:51]: `tailwindcss-radix` is new to me. Looks sweet! I might use that in `app.makeswift.com`

Comment (migueloller) [reply examples/bigcommerce/tailwind.config.js:51]: Same for me, very cool!

**Review by migueloller — APPROVED** [review]

**Pull request merged**